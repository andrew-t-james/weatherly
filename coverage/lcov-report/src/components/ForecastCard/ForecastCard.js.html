<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/ForecastCard/ForecastCard.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/ForecastCard</a> ForecastCard.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">88.89% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>8/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">92.86% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>13/14</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">88.89% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>8/9</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
import './ForecastCard.css';
&nbsp;
const ForecastCard = props =&gt; {
  const { iconTitle, high, low, conditions, day } = props.forecast;
  let { hour } = props.forecast;
&nbsp;
  <span class="missing-if-branch" title="if path not taken" >I</span>if (hour &gt; 12) {
<span class="cstat-no" title="statement not covered" >    hour = hour - 12;</span>
  } else if (hour &lt; 1) {
    hour = '00';
  }
&nbsp;
  return (
    &lt;li
      tabIndex="0"
      aria-label="Hourly Forecast Item"
      className="forecast-list-item"&gt;
      {
        hour ?
          &lt;h2
            className="forecast-list-item__hour"&gt;
            {`${hour}:00`}
          &lt;/h2&gt; :
          &lt;h2
            tabIndex="0"
            aria-label={`Weather for ${day}`}
            className="forecast-list-item__title"&gt;
            {day}
          &lt;/h2&gt;
      }
      {iconTitle &amp;&amp; &lt;i className={`forecast-list-item__icon wi wi-wu-${iconTitle}`}  alt={iconTitle} /&gt;}
      {
        conditions &amp;&amp;
        &lt;p
          tabIndex="0"
          aria-label={`Weather will be ${conditions} at ${hour} o'clock`}
          className="forecast-conditions"&gt;
          {conditions}&amp;#176;
        &lt;/p&gt;
      }
      {
        high &amp;&amp;
        &lt;p
          tabIndex="0"
          aria-label={`Today the high will be ${high} degrees`}
          className="forecast-high"&gt;
          {high} &amp;#176; &amp;uarr;
        &lt;/p&gt;
      }
      {low &amp;&amp;
      &lt;p
        tabIndex="0"
        aria-label={`Today the low will be ${low} degrees`}
        className="forecast-low"&gt;
        {low} &amp;#176; &amp;darr;
      &lt;/p&gt;}
    &lt;/li&gt;
  );
};
&nbsp;
ForecastCard.propTypes = {
  forecast: PropTypes.object,
  iconTitle: PropTypes.string,
  low: PropTypes.string,
  day: PropTypes.string,
  conditions: PropTypes.number,
  hour: PropTypes.number
};
&nbsp;
export default ForecastCard;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu May 31 2018 09:58:02 GMT-0600 (MDT)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
