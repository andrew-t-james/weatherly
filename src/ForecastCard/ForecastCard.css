.seven-day-forecast {
  border: 1px solid red;
  padding: 2rem 0;
  width: 100%;
}


.seven-day-forecast-list {
  display: flex;
  flex-direction: column;
  font-size: 1.4rem;
  justify-content: space-around;
  list-style: none;
  text-align: center;
}

@media screen and  (min-width: 560px) {
  .seven-day-forecast-list {
    flex-direction: row;
    flex-wrap: wrap;
  }
}

.seven-day-forecast-title {
  font-size: 3rem;
  margin-bottom: 2rem;
  text-align: center;
}

.seven-day-forecast-list-item {
  border: 1px solid green;
  flex: 1;
  /* flex-basis: 30rem; */
  padding: 2rem 1rem;
}

.high-low {
  display: flex;
  justify-content: space-around;
}