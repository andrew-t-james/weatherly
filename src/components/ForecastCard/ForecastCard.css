.forecast-title {
  font-size: 3rem;
  margin-bottom: 2rem;
  text-align: center;
}

.forecast-list-item {
  background-color: var(--color-grey-light-2);
  border-radius: 4px;
  box-shadow: 5px 5px 25px 0 rgba(46,61,73,0.2);
  flex: 1;
  flex-basis: 8rem;
  margin: 1rem;
  padding: 2rem 1rem;
  transition: all 0.2s ease-in-out;
}

.forecast-list-item:hover {
  box-shadow: 2px 4px 8px 0 rgba(46, 61, 73, 0.2);
}

.forecast-list-item__hour,
.forecast-list-item__title,
.forecast-conditions,
.forecast-high,
.forecast-low {
  color: var(--color-grey-dark-1);
}

.forecast-conditions,
.forecast-high,
.forecast-low {
  font-size: 1.6rem;
}

.forecast-list-item__hour {
  font-size: 2rem;
}

.forecast-list-item__title {
  font-size: 1.5rem;
}

.forecast-list-item__icon {
  font-size: 3rem;
  margin: 1rem;
}