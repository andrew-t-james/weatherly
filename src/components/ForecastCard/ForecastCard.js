import React from 'react';
import './ForecastCard.css';

const ForecastCard = (props) => {
  const { iconUrl, iconTitle, high, low, conditions, day } = props.forecast;
  let { hour } = props.forecast;

  if (hour > 12) {
    hour = hour - 12;
  } else if (hour < 1) {
    hour = '00';
  }

  return (
    <li className="forecast-list-item">
      {
        hour ?
          <h2 className="forecast-list-item__hour">{`${hour}:00`}</h2> :
          <h2 className="forecast-list-item__title">{day}</h2>
      }
      <img className="forecast-list-item__image" src={iconUrl} alt={iconTitle}/>
      {conditions && <p className="forecast-conditions">{conditions}&#176;</p>}
      <div className="high-low">
        {high && <p className="forecast-high">&uarr; {high} &#176;</p>}
        {low &&  <p className="forecast-low">{low} &#176; &darr;</p>}
      </div>
    </li>
  );
};

export default ForecastCard;