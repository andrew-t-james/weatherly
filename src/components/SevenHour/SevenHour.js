import React from 'react';
import ForecastCard from '../ForecastCard/ForecastCard.js';

const SevenHour = (props) => {
  const { hourly } = props;

  return (
    <div className="forecast">
      <h1 className="forecast-title">Seven Hour</h1>
      <ul className="forecast-list">
        {
          hourly.map((hour, index) =>
            <ForecastCard
              key={index}
              forecast={hour}/>
          )
        }
      </ul>
    </div>
  );
};

export default SevenHour;