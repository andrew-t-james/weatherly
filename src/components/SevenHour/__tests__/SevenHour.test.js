import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import SevenHour from '../SevenHour.js';
import ForecastCard from '../../ForecastCard/ForecastCard.js';

describe('Seven Hour unit tests', () => {
  const sevenHourForecast = [
    {
      conditions: 47,
      day: 12,
      high: null,
      iconTitle: "partlycloudy",
      iconUrl: "http://icons.wxug.com/i/c/k/partlycloudy.gif",
      low: null
    },
    {
      conditions: 49,
      day: 13,
      high: null,
      iconTitle: "partlycloudy",
      iconUrl: "http://icons.wxug.com/i/c/k/partlycloudy.gif",
      low: null
    }
  ];

  const sevenHour = shallow(<SevenHour hourlyForecast={sevenHourForecast} />);

  test('it should render multiple Cards components', () => {
    const expected = 2;
    const actual = sevenHour.find(ForecastCard).length;

    expect(actual).toEqual(expected);
  });

  test('it should match snapshot', () => {
    const sevenHour = renderer
      .create(<SevenHour hourlyForecast={sevenHourForecast} />)
      .toJSON();

    expect(sevenHour).toMatchSnapshot();
  });
});