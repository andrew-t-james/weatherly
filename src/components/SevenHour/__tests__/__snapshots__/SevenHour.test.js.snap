// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Seven Hour unit tests it should match snapshot 1`] = `
<section
  className="forecast"
>
  <h1
    aria-label="Hourly Forecast for the next 7 hours"
    className="forecast-title"
    tabIndex="0"
  >
    Hourly
  </h1>
  <ul
    className="forecast-list"
  >
    <li
      aria-label="Hourly Forecast Item"
      className="forecast-list-item"
      tabIndex="0"
    >
      <h2
        aria-label="Weather for 12"
        className="forecast-list-item__title"
        tabIndex="0"
      >
        12
      </h2>
      <i
        alt="partlycloudy"
        className="forecast-list-item__icon wi wi-wu-partlycloudy"
      />
      <p
        aria-label="Weather will be 47 at undefined o'clock"
        className="forecast-conditions"
        tabIndex="0"
      >
        47
        °
      </p>
    </li>
    <li
      aria-label="Hourly Forecast Item"
      className="forecast-list-item"
      tabIndex="0"
    >
      <h2
        aria-label="Weather for 13"
        className="forecast-list-item__title"
        tabIndex="0"
      >
        13
      </h2>
      <i
        alt="partlycloudy"
        className="forecast-list-item__icon wi wi-wu-partlycloudy"
      />
      <p
        aria-label="Weather will be 49 at undefined o'clock"
        className="forecast-conditions"
        tabIndex="0"
      >
        49
        °
      </p>
    </li>
  </ul>
</section>
`;
